compiler/rustc_hir_typeck/src/_match.rs

```diff
@@ -593,96 +587,35 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
-    /// When we have a `match` as a tail expression in a `fn` with a returned `impl Trait`
-    /// we check if the different arms would work with boxed trait objects instead and
-    /// provide a structured suggestion in that case.
-    pub(crate) fn opt_suggest_box_span(
+    // Does the expectation of the match define an RPIT?
+    // (e.g. we're in the tail of a function body)
```

Comment:
would probably make sense to document that the `DefId` return type is of the `AliasTy` for the expectation. A little bit unclear by the name. Maybe it should be renamed to something like `alias_ty_id_from_rpit_expectation`